src/events/logic.py

```diff
@@ -224,8 +224,9 @@ class Events:
     ON_ARTICLE_PUBLISHED = 'on_article_published'
 
     # kwargs: request, article
-    # raised when an Editor notifies an author that publication is set
-    ON_AUTHOR_PUBLICATION = 'on_author_publication'
```

Comment:
Even if we de-register the `send_author_publication_notification`, we should properly deprecate an event like this one before deletion as others might depend on it.

I'd recommend adding such warning on `events.logic.Event.register_for_event` perhaps with a set of deprecated events:
 ```python
 class Event():
 DEPRECATED_EVENTS = {
    ON_AUTHOR_PUBLICATION,
}
def register_for_event(event, *functions):
    if event in self.DEPRECATED_EVENTS:
        # raise DeprecationWarning
[...]

 ```

Reply:
Good call, but if we are trying to add a deprecation warning for this event because we think people might be using it in plugins, we need to make it work with all the supporting functions.

I've added back the full set of functions to make this old event work the way it used to, and I've added the deprecation warning in `raise_event`.